### PR TITLE
Move some HWFusedPatterns to IgnoredPatterns

### PR DESCRIPTION
**Pull request by kshpv**
### Changes

Move some patterns from HWFusedPatterns to IgnoredPatterns.

### Reason for changes

Some of the patterns inside HWFusedPatterns don't really align with the idea behind this class. These patterns were excluded from HWFusedPatterns and included in IgnoredPatterns

### Related tickets

112515

### Tests

TBD

### CODE REVIEW

File: nncf/common/graph/graph_matching.py [hunk 1]
```diff
@@ -8,22 +8,56 @@
...
+    """
+    non_pattern_nodes = [
+        node_id
+        for node_id, node_data in pattern.nodes(data=True)
+        if GraphPattern.NON_PATTERN_NODE_TYPE in node_data[GraphPattern.METATYPE_ATTR]
+    ]
+    return len(pattern) - len(non_pattern_nodes)
+
+
+def _is_subgraph_matching_strict(graph: nx.DiGraph, pattern: nx.DiGraph, subgraph: Dict[str, str]) -> bool:
```
[the remark this answers is not in the document]
Comment (kshpv) [reply nncf/common/graph/graph_matching.py:51]: When we do the subgraph isomorphism problem we need to do an extra processing step to remove subgraphs that have outgoing and ingoing edges, which this function resolves.

File: nncf/torch/quantization/ignored_patterns.py [hunk 2]
```diff
@@ -15,35 +15,80 @@
...
     pattern.add_edge(reshape, matmul)
-    pattern.add_edge(non_pattern_node_2, matmul)
+    pattern.add_edge(matmul_branch_nodes, matmul)
+    return pattern
+
+
+@PT_IGNORED_PATTERNS.register(IgnoredPatternNames.MULTIHEAD_ATTENTION_OUTPUT)
+def create_multihead_attention_output() -> GraphPattern:
+    matmul_aliases = ["linear", "addmm", "matmul", "bmm", "mm", "baddbmm"]
+    reshape_squeeze_aliases = ["reshape", "view", "flatten", "squeeze", "unsqueeze", "squeeze", "flatten", "unsqueeze"]
```
[the remark this answers is not in the document]
Comment (kshpv) [reply nncf/torch/quantization/ignored_patterns.py:75]: `reshape` and `squeeze` ops are merged into one metatype - `PTReshapeMetatype`. I saved this PT logic here

File: tests/onnx/test_pattern_manager.py [hunk 3]
```diff
@@ -23,11 +23,7 @@
     HWFusedPatternNames.NORMALIZE_L2_MULTIPLY: "Not relevant for ONNX.",
```
[the remark this answers is not in the document]
Comment (kshpv) [reply tests/onnx/test_pattern_manager.py:23]: OpenVINO supported these patterns while ONNX and Torch did not. So for OpenVINO, there is no need to update it.

File: nncf/common/graph/graph_matching.py [hunk 4]
```diff
@@ -37,119 +71,81 @@ def is_subgraph_has_inner_outgoing_edges(
...
+        if node_from_pattern in last_nodes and has_external_predcessors:
+            return False
+        if (node_from_pattern not in last_nodes and node_from_pattern not in starting_nodes) and (
+            has_external_successors or has_external_predcessors
+        ):
+            return False
+    return True
+
+
+def _copy_subgraph_excluding_non_pattern_node(subgraph: Dict[str, str], pattern_graph: GraphPattern) -> Dict[str, str]:
```
[the remark this answers is not in the document]
Comment (kshpv) [reply nncf/common/graph/graph_matching.py:104]: They don't count as pattern nodes, so they should not be included for `_is_subgraph_matching_strict`

File: nncf/openvino/quantization/ignored_patterns.py [hunk 5]
```diff
@@ -16,33 +16,63 @@
 OPENVINO_IGNORED_PATTERNS = Registry("IGNORED_PATTERNS")
 
 
-@OPENVINO_IGNORED_PATTERNS.register(IgnoredPatternNames.SOFTMAX_MATMUL)
-def softmax_matmul() -> GraphPattern:
-    pattern = GraphPattern()
+def _add_softmax_matmul(pattern: GraphPattern) -> None:
```
Comment (andrey-churkin) [inline nncf/openvino/quantization/ignored_patterns.py:19]: You removed the following fused patterns for OpenVINO backend
- HWFusedPatternNames.STABLE_DIFFUSION
- HWFusedPatternNames.SOFTMAX_RESHAPE_MATMUL
- HWFusedPatternNames.SOFTMAX_RESHAPE_TRANSPOSE_MATMUL
- HWFusedPatternNames.SOFTMAX_RESHAPE_TRANSPOSE_GATHER_MATMUL
- HWFusedPatternNames.MATMUL_SOFTMAX_MATMUL

Did you verify that everything works correctly with new ignored patterns?

Comment (kshpv) [reply nncf/openvino/quantization/ignored_patterns.py:19]: I verified with the validation on the dedicated model scope. The accuracy numbers for them are equal. Also, I mannualy check the model graphs for GPT-2, swin, roberta models. They are equal.

**Review by andrey-churkin — APPROVED** [review]

**Pull request merged**